odd-platform-api/src/main/java/com/provectus/oddplatform/repository/TagRepositoryImpl.java

```diff
@@ -40,6 +45,22 @@ public TagRepositoryImpl(final DSLContext dslContext) {
             .collect(Collectors.toList());
     }
 
+    @Override
+    public List<TagDto> listMostPopular(final String query, final int page, final int size) {
+        return dslContext
+            .select(TAG.asterisk())
+            .select(DSL.count(TAG_TO_DATA_ENTITY.TAG_ID))
+            .from(TAG)
+            .join(TAG_TO_DATA_ENTITY).on(TAG.ID.eq(TAG_TO_DATA_ENTITY.TAG_ID))
+            .where(queryCondition(query))
```

Comment:
`listCondition(query)`

It also adds soft delete filter

Reply:
Aight, thanks!